src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -26,7 +26,7 @@ private sealed class SwitchStatementLocalRewriter : BaseSwitchLocalRewriter
             /// </summary>
             private readonly Dictionary<SyntaxNode, LabelSymbol> _sectionLabels = PooledDictionary<SyntaxNode, LabelSymbol>.GetInstance();
 
-            protected override bool GenerateSequencePoints => true;
+            protected override bool GenerateInstrumentation => true;
```

Comment:
>true [](start = 63, length = 4)

Unconditional instrumentation feels suspicious

Reply:
All statements are instrumented today.  For example, this long appears (without any condition) in LocalRewriter.VisitBlock:
```csharp
            BoundStatement? prologue = _instrumenter.CreateBlockPrologue(node, out synthesizedLocal);
```

Normally we simply call the instrumentation API when lowering a statement.  Since this overrides a method in code that is shared between processing expressions (e.g. the is-pattern expression) and statmements (e.g. the switch statement) we use this to communicate which kind of context we are in.

---
In reply to: [412362045](https://github.com/dotnet/roslyn/pull/43430#discussion_r412362045) [](ancestors = 412362045)

Reply:
Normally we instrument conditionally, based on ```LocalRewriter.Instrument``` value. Even the ```LocalRewriter.VisitBlock``` that you use as an example bypasses instrumentation when ```LocalRewriter.Instrument``` is false (see an ```if``` statement 5 lines above). I expect at least that value to be always taken into account.

---
In reply to: [412452424](https://github.com/dotnet/roslyn/pull/43430#discussion_r412452424) [](ancestors = 412452424,412362045)